presto-spi/src/main/java/io/prestosql/spi/expression/Constant.java

```diff
@@ -48,7 +48,7 @@ public Object getValue()
     @Override
     public int hashCode()
     {
-        return Objects.hash(value, getType());
```

Comment:
Value itself doesn't seem enough for comparison (e.g multiple types can be represented by long). Maybe we should precalculate has value (we do that in other classes where hash computation is expensive)

Reply:
Thanks for the suggestion on caching the value, that makes sense. 

I'm curious if there's a historical reason behind not implementing/precalculating `hashcode` method in some types (ArrayType, RowType etc).

Reply:
@phd3 is this commit the gist of the change?

Reply:
@findepi I think both the commits are crucial, the hashCode issue is more important in context of `FieldDereference` class.

Reply:
> I'm curious if there's a historical reason behind not implementing/precalculating hashcode method in some types (ArrayType, RowType etc).

What do you mean specifically? If you mean particular rows, one would need to keep hashcode in Block itself, which cost memory and might not be that beneficial in general processing